plugins/woocommerce-admin/client/analytics/settings/config.js

```diff
@@ -143,6 +148,5 @@ export const config = applyFilters( SETTINGS_FILTER, {
 			'Database date field considered for Revenue and Orders reports',
 			'woocommerce'
 		),
-		defaultValue: 'date_created',
```

Comment:
We probably want to keep this and set it to `date_paid` it does feel slightly strange going to the settings screen and seeing the `Select a date type`.
But it did look like the `defaultValue` doesn't work right now, so it would have to be added here: https://github.com/woocommerce/woocommerce/blob/4c821a6b4edc889e99abce857eada3c0bcc4b5f8/plugins/woocommerce-admin/client/analytics/settings/setting.js#L82 (`value || defaultValue`)

Reply:
I removed it because I was having problems when clicking on save without touching the date field. In the case where the option was not defined, the default value would be shown, leading the user to believe that the option is set to the default value. However, when clicking on save without changing the value (touching the field), the value is not persisted (the field is not perceived as "dirty")... That's why I opted to not use defaultValue for now and set a value for the option only when the user explicitly defines it. (Even though the behavior for the reports will default to date_paid when the option is not defined).